### DISCUSSION TITLE
Ph sensor

### DISCUSSION THREAD

**Discussion by pmatvos** [theyosh/TerrariumPI#668] [Q&A] [answered]
Dose anyone know if terrariumpi 4.2 suports this probe sensor?
https://m.aliexpress.com/item/1005002842540357.html?spm=a2g0n.detail.0.0.386b5bf3b8qIBA&gps-id=platformRecommendH5&scm=1007.18499.271340.0&scm_id=1007.18499.271340.0&scm-url=1007.18499.271340.0&pvid=0f31eddb-cb28-4e7f-ac4b-5e1d74345ae5&_t=gps-id:platformRecommendH5,scm-url:1007.18499.271340.0,pvid:0f31eddb-cb28-4e7f-ac4b-5e1d74345ae5,tpp_buckets:668%232846%238109%231935&pdp_ext_f=null&pdp_pi=-1%253B1.93%253B-1%253B-1%2540salePrice%253BEUR%253Brecommend-recommend

**Comment by theyosh** [comment]
Hmm, I have not tested Ph sensors. There should be 1 working. https://theyosh.github.io/TerrariumPI/hardware/sensor/sen0161-ph-probe-sensor-analog/ And that is using: https://theyosh.github.io/TerrariumPI/hardware/#analog-sensors

So I guess that you need at least a https://rasp.io/analogzero/ to connect the sensor to the Pi. This will take some GPIO space...

But there are more ways. You could also use an Arduino to readout the Ph sensor, provide a web server on the Arduino so TP4 can read it out using it as a remote sensor.

Or if you are a bit handy and have some programming skills, make you own script/code to do a readout of the sensor. This can be done in many ways. But when you have a script, you could also use that as a sensor source, to get the Ph sensor readout.

Sorry to not have a clear answer, but this is an analog sensor, which is never directly supported by TP4

**Reply by pmatvos** [reply]
Sorry for extremely late response. Thank you for the answer. It was wery useful. So when I connect the sensor to the software i can select read out from script and import my script. I have ordered the sensor and adc chip but unfortunately it will take a couple of months before I get it. I will tell you if it work or I'll post the script here if it will be necessary.

**Reply by theyosh** [reply, marked as answer]
Yeah, I think making your own code for the readout would the easiest way for analog sensors. So, if you have that working, I can help to get it in TP4.
